File: src/1161-Maximum-Level-Sum-of-a-Binary-Tree/1161.go
func maxLevelSum(root *TreeNode) int {
    if root == nil {
        return 0
    }
    
    q := []*TreeNode{root}
    res, maxSum, level := 0, 0, 0
    for len(q) > 0 {
        curSum, qLen := 0, len(q)
        for i := 0; i < qLen; i++ {
            node := q[0]
            q = q[1:]
            curSum += node.Val
            
            if node.Left != nil {
                q = append(q, node.Left)
            }
            if node.Right != nil {
                q = append(q, node.Right)
            }
        }
        level++
        if curSum > maxSum {
            maxSum, res = curSum, level
        }
    }
    return res
}

File: src/1161-Maximum-Level-Sum-of-a-Binary-Tree/1161.cpp
class Solution 
{
public:
    int maxLevelSum(TreeNode* root) 
    {
        if (!root)   return 0;
        queue<TreeNode*> q; q.push(root);
        int maxSum = 0, res = 0, level = 0;
        
        while (q.size())
        {
            int curSum = 0, qLen = q.size();
            for (int i = 0; i < qLen; ++i)
            {
                auto node = q.front(); q.pop();
                curSum += node->val;
                if (node->left) q.push(node->left);
                if (node->right) q.push(node->right);
            }
            
            level++;
            if (curSum > maxSum) 
            {
                maxSum = curSum;
                res = level;
            }
        }
        return res;
    }
};